---
title: "Local ancestry analyses supps"
author: "Melissa Spear"
date: "13/05/2020"
output: html_document
---

The following code is for Supplementary Figures 6-9 in the paper. 

Supplementary Figure 6
```{r}
#Reading in Admx Map Results 
hchs_sol_AGE_admixture_results = read.table("/Volumes/LaCie/Downloads/home/mlspear/HCHS_SOL/data/phased_data_030718/AdmxMapping/Mexicans/HCHS_SOL_Mexicans_AdmxMap_NAM_AGE_results_AllChr_020519.txt", header= TRUE)
hchs_sol_GEN_admixture_results = read.table("/Volumes/LaCie/Downloads/home/mlspear/HCHS_SOL/data/phased_data_030718/AdmxMapping/Mexicans/HCHS_SOL_Mexicans_AdmxMap_NAM_GEN_results_AllChr_020519.txt", header= TRUE)

install.packages("coda")
library(coda)

p = hchs_sol_AGE_admixture_results$p

# Calculating the number of tests
# It will print the number of ancestry block and the new threshold for significance

b= hchs_sol_AGE_admixture_results$p

total_eff= effectiveSize(-log10(b))

print(paste('effective number of blocks is',total_eff,'or a threshold of ',0.05/total_eff) )
```

```{r}
#manhattan plot function
manhattan <- function(dataframe, colors=c("gray10", "gray50"), ymax=8, limitChromosomes=1:23, suggestiveline=-log10(1e-5), genomewideline=-log10(5e-8), annotate=NULL, ...) {

    d=dataframe
    if (!("CHR" %in% names(d) & "BP" %in% names(d) & "p" %in% names(d))) stop("Make sure your data frame contains columns CHR, BP, and P")
    
    if (any(limitChromosomes)) d=d[d$CHR %in% limitChromosomes, ]
    d=subset(na.omit(d[order(d$CHR, d$BP), ]), (p>0 & p<=1)) # remove na's, sort, and keep only 0<P<=1
    d$logp = -log10(d$p)
    d$pos=NA
    ticks=NULL
    lastbase=0
    colors <- rep(colors,max(d$CHR))[1:max(d$CHR)]
    if (ymax=="max") ymax<-ceiling(max(d$logp))
    if (ymax<8) ymax<-8
    
    numChroms=length(unique(d$CHR))
    if (numChroms==1) {
        d$pos=d$BP
        ticks=floor(length(d$pos))/2+1
    } else {
        for (i in unique(d$CHR)) {
          if (i==1) {
    			d[d$CHR==i, ]$pos=d[d$CHR==i, ]$BP
    		} else {
    			lastbase=lastbase+tail(subset(d,CHR==i-1)$BP, 1)
    			d[d$CHR==i, ]$pos=d[d$CHR==i, ]$BP+lastbase
    		}
    		ticks=c(ticks, d[d$CHR==i, ]$pos[floor(length(d[d$CHR==i, ]$pos)/2)+1])
    	}
    }
    
    if (numChroms==1) {
        with(d, plot(pos, logp, ylim=c(0,5), ylab=expression(-log[10](italic(p))), xlab=paste("Chromosome",unique(d$CHR),"position"), ...))
    }	else {
        with(d, plot(pos, logp, ylim=c(0,5), ylab=expression(-log[10](italic(p))), xlab="Chromosome", xaxt="n", type="n", ...))
        axis(1, at=ticks, lab=unique(d$CHR), ...)
        icol=1
        for (i in unique(d$CHR)) {
            with(d[d$CHR==i, ],points(pos, logp, col=colors[icol], ...))
            icol=icol+1
    	}
    }
    
    if (!is.null(annotate)) {
        d.annotate=d[which(d$SNP %in% annotate), ]
        with(d.annotate, points(pos, logp, col="green3", ...)) 
    }
    
    if (suggestiveline) abline(h=suggestiveline, col="blue")
    if (genomewideline) abline(h=genomewideline, col="red")
    #abline(7.30,0,lty=3,col="black")
    abline(3.86,0,lty=3,col="black") #for admixture mapping 
}

```


```{r}
#merge of files 
map_file = read.table("/Volumes/LaCie/Downloads/home/mlspear/HCHS_SOL/data/phased_data_030718/RFMix/ALL_Chr.predicted.map", header=F)
names(map_file) = c("CHR","Probe","cM", "BP")

final_admixture_results_AGE= merge(map_file, hchs_sol_AGE_admixture_results, by = 'Probe')
final_admixture_results_GEN= merge(map_file, hchs_sol_GEN_admixture_results, by = 'Probe')

x = hchs_sol_AGE_admixture_results[hchs_sol_AGE_admixture_results$p <= "0.00013798",]
y = hchs_sol_GEN_admixture_results[hchs_sol_GEN_admixture_results$p <= "0.00013798",]

require(RColorBrewer)
pcol=brewer.pal(8, "Set2")
```

```{r}
#plot 6a 
#png("/Users/melissaspear/Dropbox/HCHS_SOL/paper/figures/HCHS_SOL_Mexicans_AdmxMap_manhattan_AGE_051519_wide_line.png", width = 800, height = 400)
manhattan(final_admixture_results_AGE, colors=c(pcol[1],pcol[3],pcol[8]), pch=20, genomewideline=F, suggestiveline=F)
#dev.off()

```

```{r}
#plot 6b 
#png("/Users/melissaspear/Dropbox/HCHS_SOL/paper/figures/HCHS_SOL_Mexicans_AdmxMap_manhattan_GEN_051519_wide_line.png", width = 800, height = 400)
manhattan(final_admixture_results_GEN, colors=c(pcol[1],pcol[3],pcol[8]), pch=20, genomewideline=F, suggestiveline=F)
#dev.off()
```


Supplementary Figure 7 
